test/integration/client/client_test.go

```diff
@@ -1342,3 +1355,192 @@ func TestExtractModifyApply_ForceOwnership(t *testing.T) {
 		t.Errorf("createMgrExtracted apply configuration did not match expected, got:\n%s\n", cmp.Diff(expectedCreateExtracted, createMgrExtracted))
 	}
 }
+
+func TestAPIServerTransportMetrics(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllAlpha", true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllBeta", true)()
+
+	result := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--disable-admission-plugins", "ServiceAccount"}, framework.SharedEtcd())
+	defer result.TearDownFn()
+
+	client := clientset.NewForConfigOrDie(result.ClientConfig)
+	dynamicClient, err := dynamic.NewForConfig(result.ClientConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ns := framework.CreateNamespaceOrDie(client, "test-api-server-metrics", t)
+
+	// Create one resource for each registered type
+	createData := etcd.GetEtcdStorageData()
+
+	// gather resources to test
+	_, resourceLists, err := client.Discovery().ServerGroupsAndResources()
+	if err != nil {
+		t.Fatalf("Failed to get ServerGroupsAndResources with error: %+v", err)
+	}
+
+	for _, resourceList := range resourceLists {
+		for _, resource := range resourceList.APIResources {
+			if !strings.HasSuffix(resource.Name, "/status") {
```

Comment:
Is this backwards?

```suggestion
			if strings.HasSuffix(resource.Name, "/status") {
```

Reply:
I copied from one test, it seems that does check status only, good catch

Reply:
if I change this line the test fails, I need to understand the logic it uses now 🙃

Reply:
I'm not sure this really makes a difference, I already have a test that verifies the client-go generated clientset shares a transport

Reply:
I am not sure I understand - why does the test fail?  It is not clear to me why you need the if statement at all?

Reply:
Question: would it make the metrics results easier to reason about if you did:

1. Create/delete everything
2. Global metrics delete via the equivalent REST call `kubectl delete --raw /metrics`
3. Create/delete everything again

At that point the client numbers should be stable enough to require like a 99% cache hit rate?

This may be getting too complex so I am fine with whatever you think is sufficient.

Reply:
I don't see any difference with and without this code, the apiserver--etcd clients are not using this cache, and the client -- apiserver are already tested, since we changed the generator to share the transport for the generated clients, I'm going to remove this code, is complex to understand and makes the test harder to rread

Reply:
oh, I replied and I didn't see your answers

> I am not sure I understand - why does the test fail? It is not clear to me why you need the if statement at all?

I don't know, if I change that it fails, but this code is copied verbatim from 
https://github.com/kubernetes/kubernetes/blob/6e541a6da7e1731730c71d1e723efbca5025330c/test/integration/apiserver/apply/status_test.go#L125-L129

so it may be possible there are some assumptions about the status

> Global metrics delete via the equivalent REST call kubectl delete --raw /metrics
> At that point the client numbers should be stable enough to require like a 99% cache hit rate?

there are a lot of controllers working on the background and I don't want to make assumptions that can generate flakiness in the future,

Reply:
Note that KCM isn't running during integration tests so only a small subset of containers are running.  But yeah it's fine to drop unnecessary complexity.